assets/blocks/quiz/category-question-block/category-question-settings.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * WordPress dependencies
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+import { Notice, PanelBody, SelectControl } from '@wordpress/components';
+import { __, _n, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import NumberControl from '../../editor-components/number-control';
+import { useQuestionCategories } from '../question-categories';
+
+/**
+ * Category question block settings controls.
+ *
+ * @param {Object}   props                    Block props.
+ * @param {Object}   props.attributes         Block attributes.
+ * @param {Object}   props.attributes.options Block options attribute.
+ * @param {Function} props.setAttributes      Update block attributes.
+ */
+const CategoryQuestionSettings = ( {
+	attributes: { options = {} },
+	setAttributes,
+} ) => {
+	const setOptions = ( next, otherAttributes = {} ) =>
+		setAttributes( {
+			...otherAttributes,
+			options: { ...options, ...next },
+		} );
+
+	const [
+		questionCategories,
+		getQuestionCategoryById,
+	] = useQuestionCategories();
+
+	const categoryOptions = [
+		{
+			value: '',
+			label: __( 'Category', 'sensei-lms' ),
+		},
+		...( questionCategories || [] ).map( ( questionCategory ) => ( {
+			value: questionCategory.id,
+			label: questionCategory.name,
+		} ) ),
+	];
+
+	const questionCategory = getQuestionCategoryById( options.category );
+
+	return (
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Category Question Settings', 'sensei-lms' ) }
+				initialOpen={ true }
+			>
+				{ ! categoryOptions.length && (
+					<Notice status="warning" isDismissible={ false }>
+						{ __( 'No question categories exist.', 'sensei-lms' ) }
+					</Notice>
+				) }
+				{ categoryOptions.length > 0 && (
+					<>
+						<SelectControl
+							label={ __( 'Category', 'sensei-lms' ) }
+							options={ categoryOptions }
+							value={ options.category }
+							onChange={ ( nextCategory ) => {
+								nextCategory = parseInt( nextCategory, 10 );
```

Comment:
Something like `categoryId` could clear things up here a bit. Also there is a duplicate `parseInt` in the setOptions call below.
Bonus the ultimate JS way of casting to number: `+nextCategory`

Reply:
TIL `+nextCategory` 45d1a60b184fc97c9b5ef4d1f5f76a06a8a130d9

Reply:
Not sure what happened to comment. Fixed in https://github.com/Automattic/sensei/pull/4080/commits/45d1a60b184fc97c9b5ef4d1f5f76a06a8a130d9